### PR TITLE
Do not start application context automatically

### LINKED ISSUE #525: Do not start the context automatically
In most cases it is a good thing to start the other beans as part of the binding process, but in some cases there are undesirable side effects.

See: https://github.com/spring-projects/spring-integration-java-dsl/issues/80#issuecomment-218245617

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- This will avoid other lifecycle beans inside the application context starting when
binding lifecycle beans are started.

Resolves #525

### CODE REVIEW

File: spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc [hunk 1]
```diff
@@ -387,6 +387,12 @@ public class SendingBean {
 }
 ----
 
+===== Spring application context for Spring Cloud Stream applications
```
Comment (mbogoevici) [inline spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: Not sure if we need to add this as a documentation feature. Probably part of the upgrade guide.

**Comment by mbogoevici** [comment]
This looks incomplete - `org.springframework.cloud.stream.binder.LifecycleBinderTests` specifically tests that a regular lifecycle is started. If it passes, it means that the issue is not fixed. If it doesn't pass, it needs to be removed.

**Comment by ilayaperumalg** [comment]
It looks like the test `testNonSmartLifecyclesStarted` is incorrectly asserts [here] (https://github.com/spring-cloud/spring-cloud-stream/blob/master/spring-cloud-stream/src/test/java/org/springframework/cloud/stream/binder/LifecycleBinderTests.java#L42) and after modifying it to use `assertThat(simpleLifecycle.isRunning()).isTrue();` the test fails as expected :-) 

Do you think we can still keep this test with the assertion on application context not started for the explicit lifecycle bean?

**Comment by mbogoevici** [comment]
Oh, that must have been broken during migration to AssertJ - we might want to keep that around, and modify the assert to False.

File: spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc (hunk 1 above)
Comment (ilayaperumalg) [inline spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: I am not sure where is `upgrade` guide though ;)

**Comment by mbogoevici** [comment]
We should add a section with 'What's new in 1.1' under intro.

**Comment by sobychacko** [comment]
Merged.

**Pull request closed without merging**